action/protocol/staking/protocol.go

```diff
@@ -601,12 +602,14 @@ func (p *Protocol) settleAction(
 	return &r, nil
 }
 
-func (p *Protocol) needToReadCandsMap(height uint64) bool {
-	return height > p.config.PersistStakingPatchBlock
+func (p *Protocol) needToReadCandsMap(ctx context.Context, height uint64) bool {
```

Comment:
IMO, `needToReadCandsMap` and `needToWriteCandsMap` can be combined together due to the same code. But current implementation is fine.

Reply:
resolved offline